src/platform-aws.c

```diff
@@ -165,26 +126,47 @@ static struct ec2_platform_data *get_platform_data()
 	static struct ec2_platform_data *platform_data = NULL;
 	const size_t platform_n = sizeof(platform_data_map)/sizeof(platform_data_map[0]);
 	const char* platform_type = NULL;
+	regex_t regex;
+	int ret;
 
 	nccl_net_ofi_mutex_lock(&mutex);
 
 	if (init) {
-		nccl_net_ofi_mutex_unlock(&mutex);
-		return platform_data;
+		goto done;
 	}
 	init = true;
 
 	platform_type = nccl_net_ofi_get_product_name();
 	if (platform_type == NULL) {
-		nccl_net_ofi_mutex_unlock(&mutex);
-		return NULL;
+		platform_data = NULL;
+		goto done;
 	}
 
 	for (size_t idx = 0; idx < platform_n; idx++) {
-		if (strcmp(platform_type, platform_data_map[idx].name) == 0)
+		ret = regcomp(&regex, platform_data_map[idx].name, 0);
+		if (ret != 0) {
+			NCCL_OFI_WARN("Could not compile platform_type regex for %s",
+				      platform_data_map[idx].name);
+			platform_data = NULL;
+			goto done;
+		}
+
+		ret = regexec(&regex, platform_type, 0, NULL, 0);
+
+		if (ret == 0) {
 			platform_data = &platform_data_map[idx];
```

Comment:
break out of the loop on first hit?

Reply:
I should really not write code in a hurry.  Fixed (and moved the regfree() up to avoid the leak).